openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java

```diff
@@ -102,4 +103,20 @@ void importSaveForScenario(@Param("id") String id,
   @Override
   @Query("select count(distinct i) from Inject i where i.createdAt < :creationDate")
   long globalCount(@Param("creationDate") Instant creationDate);
+
+  @Query(value = " SELECT injects.inject_id, " +
+          "coalesce(array_agg(it.team_id) FILTER ( WHERE it.team_id IS NOT NULL ), '{}') as inject_teams, " +
+          "coalesce(array_agg(assets.asset_id) FILTER ( WHERE assets.asset_id IS NOT NULL ), '{}') as inject_assets, " +
+          "coalesce(array_agg(iag.asset_group_id) FILTER ( WHERE iag.asset_group_id IS NOT NULL ), '{}') as inject_asset_groups, " +
+          "coalesce(array_agg(ie.inject_expectation_id) FILTER ( WHERE ie.inject_expectation_id IS NOT NULL ), '{}') as inject_expectations " +
+          "FROM injects " +
+          "LEFT JOIN injects_teams it ON injects.inject_id = it.inject_id " +
+          "LEFT JOIN injects_assets ia ON injects.inject_id = ia.inject_id " +
+          "LEFT JOIN injects_asset_groups iag ON injects.inject_id = iag.inject_id " +
+          "LEFT JOIN asset_groups_assets aga ON aga.asset_group_id = iag.asset_group_id " +
+          "LEFT JOIN assets ON assets.asset_id = ia.asset_id OR aga.asset_id = assets.asset_id " +
+          "LEFT JOIN injects_expectations ie ON injects.inject_id = ie.inject_id " +
```

Comment:
I am not sure that we need to retrieve asset from asset group

Reply:
Unfortunatelly, we do. It's used in AtomicTestingUtils.java line 103 (and more after this line)